lucene/core/src/java/org/apache/lucene/search/KnnCollector.java

```diff
@@ -85,4 +85,95 @@ public interface KnnCollector {
    * @return The collected top documents
    */
   TopDocs topDocs();
+
+  /**
+   * This method returns a {@link DocIdSetIterator} over entry points that seed the KNN search, or
+   * {@code null} (default) to perform a full KNN search (without seeds).
+   *
+   * <p>Note that the entry points should represent ordinals, rather than true document IDs.
+   *
+   * @return the seed entry points or {@code null}.
+   * @lucene.experimental
+   */
+  default DocIdSetIterator getSeedEntryPoints() {
+    return null;
+  }
+
+  /**
+   * KnnCollector.Decorator is the base class for decorators of KnnCollector objects, which extend
+   * the object with new behaviors.
+   *
+   * @lucene.experimental
+   */
+  public abstract static class Decorator implements KnnCollector {
+    private KnnCollector collector;
```

Comment:
```suggestion
    final private KnnCollector collector;
```

Reply:
or could be `protected` if deriving classes might need access
```suggestion
    final protected KnnCollector collector;
```